### ISSUE TITLE
Simulating change on checkbox

### ISSUE THREAD

**Issue by giidox** [japgolly/scalajs-react#296]
Is there a solution in scalajs-react to simulate change on input.checkbox? `ChangeEventData` supports `value`, but checkbox needs `checked`.

My workaround is to use my own class copy-pasted from `ChangeEventData`:
```
case class CheckboxEventData(checked: Boolean = false,
                             defaultPrevented: Boolean = false) {
  def toJs: Object = {
    val t = Dynamic.literal()
    t.updateDynamic("defaultPrevented")(defaultPrevented)
    t.updateDynamic("checked")(checked)
    val o = Dynamic.literal("target" -> t)
    o
  }
  def simulate(t: ReactOrDomNode) = ReactTestUtils.Simulate.change(t, toJs)
  def simulation = Simulation.change(toJs)
}
```

And:

```CheckboxEventData(checked = true).simulate(checkbox)```

**Comment by japgolly** [comment]
I just use `Simulate.change(component)` without specifying `true` or `false`. In fact the way I write my checkbox components the `onChange` handlers don't even look at the `.target.checked` value.

I think we could just add `checked: Boolean` to `ChangeEventData` without creating a new case class.

**Comment by giidox** [comment]
Ok, thanks.

I also switched to not using .target.checked as that seems a cleaner way (dispatching an action with !props.isChecked).

**Comment by japgolly** [comment]
Closed via 31a524c
